client/my-sites/checkout/checkout-thank-you/jetpack-checkout-siteless-thank-you-completed.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import React, { FC } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+import { Card } from '@automattic/components';
+
+/**
+ * Internal dependencies
+ */
+import Main from 'calypso/components/main';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import {
+	isProductsListFetching as getIsProductListFetching,
+	getProductName,
+} from 'calypso/state/products-list/selectors';
+import JetpackLogo from 'calypso/components/jetpack-logo';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface Props {
+	productSlug: string | 'no_product';
+}
+
+const JetpackCheckoutSitelessThankYouCompleted: FC< Props > = ( { productSlug } ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const hasProductInfo = productSlug !== 'no_product';
+
+	const productName = useSelector( ( state ) =>
+		hasProductInfo ? getProductName( state, productSlug ) : null
+	) as string | null;
```

Comment:
I'm always a little suspicious of using `as` because it can hide subtle bugs. IMO they should only be used as a last resort.

I see that `getProductName` has a jsdoc annotation already with [a return value](https://github.com/Automattic/wp-calypso/blob/e3de25f37b98f73da1121cd6cb4ea26948836277/client/state/products-list/selectors/get-product-name.js#L13) of `{?string}`. If that's not working correctly, then you could modify that jsdoc to be `{string|undefined}`. TS should be able to automatically detect the `null` here, so that this should get its type automatically created as `string | undefined | null`.